### PR TITLE
Refactor claim generation

### PR DESCRIPTION
**Pull request by tothtamas28**

### CODE REVIEW

File: kevm-pyk/src/kevm_pyk/solc_to_k.py [hunk 1]
```diff
@@ -294,72 +294,73 @@ def solc_compile(contract_file: Path, profile: bool = False) -> Dict[str, Any]:
...
+    ]
+    return KApply(contract_function_application_label, [KApply(contract.klabel), KApply(klabel, args)])
+
+
+def _callvalue_cell(method: Contract.Method) -> KInner:
+    return (
+        intToken(0)
+        if not method.payable
+        else abstract_term_safely(KVariable('_###CALLVALUE###_'), base_name='CALLVALUE')
+    )
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/solc_to_k.py:340]: Can this become a member of `Contract.Method`?

File: kevm-pyk/src/kevm_pyk/solc_to_k.py [hunk 2]
```diff
@@ -294,72 +294,73 @@ def solc_compile(contract_file: Path, profile: bool = False) -> Dict[str, Any]:
...
+
+
+def _calldata_cell(contract: Contract, method: Contract.Method) -> KInner:
+    contract_function_application_label = contract.klabel_method
+    klabel = method.production.klabel
+    assert klabel is not None
+    args = [
+        abstract_term_safely(KVariable('_###SOLIDITY_ARG_VAR###_'), base_name=f'V{name}') for name in method.arg_names
+    ]
+    return KApply(contract_function_application_label, [KApply(contract.klabel), KApply(klabel, args)])
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/solc_to_k.py:332]:
```suggestion
def _calldata_cell(contract: Contract, method: Contract.Method) -> KInner:
    contract_function_application_label = contract.klabel_method
    return KApply(contract_function_application_label, [KApply(contract.klabel), method.application])
``` 

Then this ^^^ can likely just be inlined where it is used.

Then add to class `Contract.Method`:

```
@property
def application(self: Contract.Method) -> KApply:
    klabel = self.production.klabel
    assert klabel is not None
    args = [
        abstract_term_safely(KVariable('_###SOLIDITY_ARG_VAR###_'), base_name=f'V{name}') for name in self.arg_names
    ]
    return KApply(klabel, args)
```

File: kevm-pyk/src/kevm_pyk/solc_to_k.py [hunk 3]
```diff
@@ -294,72 +294,73 @@ def solc_compile(contract_file: Path, profile: bool = False) -> Dict[str, Any]:
...
+    return CTerm(init_term).add_constraint(mlEqualsTrue(KApply('_==Int_', [dst_failed_prev, KToken('0', 'Int')])))
+
+
+def _init_term_with_calldata(
+    init_term: KInner,
+    calldata: KInner,
+    callvalue: KInner,
+) -> KInner:
+    subst = {'CALLDATA_CELL': calldata, 'CALLVALUE_CELL': callvalue}
+    return substitute(init_term, subst)
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/solc_to_k.py:363]: Seems like this doesn't need its own function.

**Review by ehildenb — APPROVED** [review]

File: kevm-pyk/src/kevm_pyk/solc_to_k.py (hunk 1 above)
Comment (tothtamas28) [reply kevm-pyk/src/kevm_pyk/solc_to_k.py:340]: Done.

File: kevm-pyk/src/kevm_pyk/solc_to_k.py (hunk 2 above)
Comment (tothtamas28) [reply kevm-pyk/src/kevm_pyk/solc_to_k.py:332]: Done.

File: kevm-pyk/src/kevm_pyk/solc_to_k.py (hunk 3 above)
Comment (tothtamas28) [reply kevm-pyk/src/kevm_pyk/solc_to_k.py]: Done.

**Pull request merged**